R/validate_per.R

```diff
@@ -20,4 +20,28 @@ validate_per <- function(per, missing_per) {
       stop("`per = '", per, "'` but must be one of 'year', 'fortnight', or 'quarter'. ")
     }
   }
+  
+  if (per == "fortnight") {
```

Comment:
Best to use `switch(per, ...)` rather than `if (switch == ...) else if `

Reply:
Will switch work if I have nested ifs like the following:
```r
  if (per == "fortnight") {
    if (is.null(.fortnights_per_yr)) {
      warning("`.fortnights_per_year` not set. Using 26 as default.")
      26
    } else if (is.numeric(.fortnights_per_yr)) {
      .fortnights_per_yr
    } else {
      stop("`per` was type '", typeof(per), "', but must be numeric.")
    } 
    
  } else if(per == "quarter") {
    4
  } else if(per == "year") {
    1
  }
```